src/MSBuild.Conversion.SDK/SemanticVersion.cs

```diff
@@ -0,0 +1,221 @@
+                {
+                    //If the prerelease section marker is not the last character in the string, take off everything after it
+                    //  and use it for the prerelease field
+                    if (prereleaseStart < value.Length - 1)
+                    {
+                        ver.PrereleaseVersion = value.Substring(prereleaseStart + 1);
+                    }
+
+                    value = value.Substring(0, prereleaseStart);
+                }
+            }
+            //If the build metadata wasn't the last metadata section found, check to see if a prerelease section exists.
+            //  If it doesn't, then neither section exists
+            else if (prereleaseStart > -1)
+            {
+                //If the prerelease version marker is not the last character in the string, take off everything after it
+                //  and use it for the prerelease version field
+                if (prereleaseStart < value.Length - 1)
+                {
+                    ver.PrereleaseVersion = value.Substring(prereleaseStart + 1);
+                }
+
+                value = value.Substring(0, prereleaseStart);
+
+                //If the build metadata section is found, extract it
+                if (buildMetadataStart > -1)
+                {
+                    //If the build metadata marker is not the last character in the string, take off everything after it
+                    //  and use it for the build metadata field
+                    if (buildMetadataStart < value.Length - 1)
+                    {
+                        ver.BuildMetadata = value.Substring(buildMetadataStart + 1);
+                    }
+
+                    value = value.Substring(0, buildMetadataStart);
+                }
+            }
+
+            string[] versionParts = value.Split('.');
+
+            if (versionParts.Length > 0)
+            {
+                int major;
+                int.TryParse(versionParts[0].TrimStart(LeadingVersionChars), out major);
+                ver.Major = major;
+            }
+
+            if (versionParts.Length > 1)
+            {
+                int minor;
+                int.TryParse(versionParts[1], out minor);
+                ver.Minor = minor;
+            }
+
+            if (versionParts.Length > 2)
+            {
+                int patch;
+                int.TryParse(versionParts[2], out patch);
+                ver.Patch = patch;
+            }
+
+            // setup the componentized versions, for later comparison. Nulls are ok, SemanticVersionComponentizedString handles them.
+            ver.BuildMetadataComponents = new SemanticVersionComponentizedString(ver.BuildMetadata);
+            ver.PrereleaseVersionComponents = new SemanticVersionComponentizedString(ver.PrereleaseVersion);
+
+            return ver;
+        }
+
+        public int CompareTo(SemanticVersion other)
+        {
+            if (other == null)
```

Comment:
```suggestion
            if (other is null)
```

we use `is null` in this house